### PR TITLE
[CBRD-23702] Modernize Heap Layers

### PR DESCRIPTION
**Pull request by alexstanro**
http://jira.cubrid.org/browse/CBRD-23702

Modernize Heap Layers
- replaced allocators - stack and fixed : used the implementation from Emery Berger
- preserved old Lea allocator
- removed unused allocators and related code from Emery Berger(Kingsley, Lea...)

Details:
COMMIT 1 : Added new heap layer from Emery Berger (containing stack and fixed allocators)
COMMIT 2 : Filter out unused files (allocators and related) of Emery Berger heaplayer
COMMIT 3 : Removed old heaplayer
COMMIT 4 : Renamed new heap layer
COMMIT 5 : Added our lea_heap and custom_heaps files
COMMIT 6:  Small fixes :  stackheap, zoneheap, lockedheap - added reset method, fixed warnings, fixed compile issues

We didn't added statistics since there is no major impact.

Require intensive testing.

### CODE REVIEW

File: src/heaplayers/customheaps.h [hunk 1]
```diff
@@ -11,18 +11,11 @@ extern void *hl_fixed_alloc (UINTPTR heap_id, size_t sz);
 extern void hl_fixed_free (UINTPTR heap_id, void *ptr);
 
 extern UINTPTR hl_register_ostk_heap (int chunk_size);
-extern void hl_clear_ostk_heap (UINTPTR heap_id);
 extern void hl_unregister_ostk_heap (UINTPTR heap_id);
 extern void *hl_ostk_alloc (UINTPTR heap_id, size_t sz);
 extern void *hl_ostk_realloc (UINTPTR heap_id, void *ptr, size_t sz);
```
Comment (applys76) [inline src/heaplayers/customheaps.h:16]: Is there any use of this interface in our recent code or in the near future?
--> hl_ostk_realloc

Comment (applys76) [inline src/heaplayers/customheaps.h:16]: Is there any use of this interface in our recent server side code or in the near future?
--> hl_ostk_realloc

Comment (alexstanro) [reply src/heaplayers/customheaps.h:16]: I believe no.

Comment (applys76) [reply src/heaplayers/customheaps.h:16]: then, is there any reason intended to remain above interface?

Comment (alexstanro) [reply src/heaplayers/customheaps.h]: Ok. I removed. Who will needs, must reintroduce it.

**Pull request merged**